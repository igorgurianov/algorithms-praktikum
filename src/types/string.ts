import { ElementStates } from "./element-states";

export type TStringElement = {
  value: string;
  state: ElementStates;
};

export type TSortingElement = {
  value: number;
  state: ElementStates;
};
