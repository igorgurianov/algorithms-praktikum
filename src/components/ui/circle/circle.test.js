import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("CIRCLE", () => {
  test("render w/o letter", () => {
    const el = renderer.create(<Circle />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ letter", () => {
    const el = renderer.create(<Circle letter="Q" />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ letter head", () => {
    const el = renderer.create(<Circle head={0} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ react el head", () => {
    const el = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ text tail", () => {
    const el = renderer.create(<Circle tail="Q" />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ react el tail", () => {
    const el = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ react el tail", () => {
    const el = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ index", () => {
    const el = renderer.create(<Circle index={0} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ small", () => {
    const el = renderer.create(<Circle isSmall />).toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ default state", () => {
    const el = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ changing state", () => {
    const el = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(el).toMatchSnapshot();
  });
  test("render w/ modified state", () => {
    const el = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(el).toMatchSnapshot();
  });
});
