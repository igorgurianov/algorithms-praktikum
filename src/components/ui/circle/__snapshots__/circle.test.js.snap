// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CIRCLE render w/ changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   changing"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`CIRCLE render w/ letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    >
      Q
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ letter head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  >
    0
  </div>
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   modified"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ react el head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circle-head"
      />
      <div
        className="circle   default"
        data-testid="circle_styles"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle-letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circle-tail"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ react el tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circle-head"
      />
      <div
        className="circle   default"
        data-testid="circle_styles"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle-letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circle-tail"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`CIRCLE render w/ react el tail 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circle-head"
      />
      <div
        className="circle   default"
        data-testid="circle_styles"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle-letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circle-tail"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`CIRCLE render w/ small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle  small default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE render w/ text tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Q
  </div>
</div>
`;

exports[`CIRCLE render w/o letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle_styles"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-tail"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
